// react-material-music-player/dist/components/TrackDetials.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = TrackDetails;
var _react = _interopRequireDefault(require("react"));
var _reactRedux = require("react-redux");
var _Box = _interopRequireDefault(require("@mui/material/Box"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function TrackDetails(props) {
  const {
    title,
    artist
  } = (0, _reactRedux.useSelector)( /**@type {import("../redux/types").useSelectCb} */
  state => {
    let currentTrack = state.playlist[state.currentTrack];
    return {
      title: currentTrack === null || currentTrack === void 0 ? void 0 : currentTrack.title,
      artist: currentTrack === null || currentTrack === void 0 ? void 0 : currentTrack.artist
    };
  }, _reactRedux.shallowEqual);
  const sx = props.sx;
  return /*#__PURE__*/_react.default.createElement(_Box.default, {
    sx: sx
  }, /*#__PURE__*/_react.default.createElement(_Box.default, {
    sx: {
      typography: "subtitl3",
      whiteSpace: "nowrap",
      overflow: "hidden"
    }
  }, title), /*#__PURE__*/_react.default.createElement(_Box.default, {
    sx: {
      typography: "subtitle2",
      whiteSpace: "nowrap",
      overflow: "hidden"
    }
  }, artist));
}
