// react-material-music-player/dist/components/VolumeControl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = VolumeControl;
var _Box2 = _interopRequireDefault(require("@mui/material/Box"));
var _VolumeUp2 = _interopRequireDefault(require("@mui/icons-material/VolumeUp"));
var _Slider2 = _interopRequireDefault(require("@mui/material/Slider"));
var _IconButton2 = _interopRequireDefault(require("@mui/material/IconButton"));
var _VolumeDown2 = _interopRequireDefault(require("@mui/icons-material/VolumeDown"));
var _react = _interopRequireDefault(require("react"));
var _reactRedux = require("react-redux");
var _actionCreators = _interopRequireDefault(require("../redux/actionCreators"));
var _withoutPropagation = _interopRequireDefault(require("../utils/withoutPropagation"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function VolumeControl(props) {
  const sx = props.sx;
  const dispatch = (0, _reactRedux.useDispatch)();
  const onVolumeChange = value => dispatch(_actionCreators.default.setVolume(value));
  const value = (0, _reactRedux.useSelector)( /**@type {import("../redux/types").useSelectCb} */
  state => state.volume);
  const handleSliderChange = (event, newValue) => {
    onVolumeChange(newValue);
  };
  return /*#__PURE__*/_react.default.createElement(_Box2.default, {
    sx: {
      display: "flex",
      direction: "row",
      wrap: "nowrap",
      alignItems: "center",
      "& > .children": {
        mx: 1
      },
      ...sx
    }
  }, /*#__PURE__*/_react.default.createElement(_IconButton2.default, {
    className: "children",
    onClick: (0, _withoutPropagation.default)(onVolumeChange, value < 10 ? 0 : value - 10),
    size: "large"
  }, /*#__PURE__*/_react.default.createElement(_VolumeDown2.default, null)), /*#__PURE__*/_react.default.createElement(_Slider2.default, {
    className: "children",
    value: value,
    "aria-labelledby": "continuous-slider",
    onChange: handleSliderChange
  }), /*#__PURE__*/_react.default.createElement(_IconButton2.default, {
    className: "children",
    onClick: (0, _withoutPropagation.default)(onVolumeChange, value > 90 ? 100 : value + 10),
    size: "large"
  }, /*#__PURE__*/_react.default.createElement(_VolumeUp2.default, null)));
}
