// react-material-music-player/dist/redux/actionCreators.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _types = require("./types");
const ActionCreators = {
  changeTrack: index => ({
    type: _types.ActionTypes.CHANGE_TRACK,
    payload: {
      index: index
    }
  }),
  play: () => ({
    type: _types.ActionTypes.PLAY
  }),
  pause: () => ({
    type: _types.ActionTypes.PAUSE
  }),
  stop: () => ({
    type: _types.ActionTypes.STOP
  }),
  updatePlaylist: playlist => ({
    type: _types.ActionTypes.UPDATE_PLAYLIST,
    payload: {
      playlist: playlist
    }
  }),
  shuffle: bool => ({
    type: _types.ActionTypes.SHUFFLE,
    payload: {
      shuffle: bool
    }
  }),
  setCurrentTime: currentTime => ({
    type: _types.ActionTypes.SET_CURRENT_TIME,
    payload: {
      currentTime: currentTime
    }
  }),
  setTimeLeft: timeLeft => ({
    type: _types.ActionTypes.SET_TIME_LEFT,
    payload: {
      timeLeft: timeLeft
    }
  }),
  seek: progress => ({
    type: _types.ActionTypes.SEEK,
    payload: {
      progress: progress
    }
  }),
  setVolume: volume => ({
    type: _types.ActionTypes.CHANGE_VOLUME,
    payload: {
      volume: volume
    }
  }),
  setRepeatMode: mode => ({
    type: _types.ActionTypes.SET_REPEAT_MODE,
    payload: {
      mode: mode
    }
  }),
  skipNext: () => ({
    type: _types.ActionTypes.SKIP_NEXT
  }),
  skipPrev: () => ({
    type: _types.ActionTypes.SKIP_PREV
  })
};
var _default = exports.default = ActionCreators;
